device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -173,9 +213,61 @@ public void close() throws IOException
      * multiplexed connection.
      * <p>
      * Users should use {@link #registerDeviceClients(Iterable)} for registering multiple devices as it has some
-     * performance improvements over repeatedly calling this method. This method blocks on each registration, whereas
+     * performance improvements over repeatedly calling this method for individual device registrations. This method blocks on each registration, whereas
+     * {@link #registerDeviceClients(Iterable)} blocks on all of the registrations after starting them all asynchronously.
+     * <p>
+     * A device client can be unregistered using {@link #unregisterDeviceClient(DeviceClient)}, {@link #unregisterDeviceClient(DeviceClient, long)},
+     * {@link #unregisterDeviceClients(Iterable)}, or {@link #unregisterDeviceClients(Iterable, long)}. A device client will not be unregistered
+     * automatically if it encounters a non-retryable exception, so users are responsible for unregistering a device client
```

Comment:
Q - "A device client will not be unregistered automatically if it encounters a non-retryable exception" - is this statement applicable only during the registration call, or during device operation as well?
i.e. -> encounter a non-retryable exception during registration vs during sending telemetry.

Reply:
It is applicable at all times. The only time when a device becomes unregistered is when the user calls multiplexingClient.unregister()